src/components/views/EditUserPage.js

```diff
@@ -1,5 +1,128 @@
+import {Button} from "../ui/Button";
+import {useHistory} from "react-router-dom";
+import Select from "react-select";
+
+/*
+Edit User Info Page
+ */
+
+//To get input for the new username
+const FormField = props => {
+    return (
+        <div className="registration field">
+            <label>
+                {props.label}
+            </label>
+            <input
+                placeholder="New Username"
+                value={props.value}
+                onChange={e => props.onChange(e.target.value)}
+                type={props.type}
+            />
+        </div>
+    );
+};
+
+//For the gender Picker
+const genderOptions = [
+    {value: 'MALE', label: 'Male'},
+    {value:  'FEMALE', label: 'Female'},
+    {value: 'OTHER', label: 'Other'}
+]
+
+
+const EditUserPage = () =>{
+    const history = useHistory();
+    const[user, setUser] = useState(null)
+    const [gender, setGender] = useState(null)
+    const [username, setUsername] = useState(null)
+
+    useEffect(() => {
+        //get the User (yourself) to display current name + gender
+        async function getUser(){
+            const id = localStorage.getItem("id")
+            const response = await api.get('/users/'+id)
+            setUser(response.data)
+            setGender(user.gender)
+            setUsername(user.gender)
+            console.log(response)
+        }
+        getUser()
+    },[]);
+
+
+    //If the getUser is too slow, placeholder
+    //Shouldn't get displayed, but otherwise <EditPlayerProfile> tries to render before user gets received
+    let editProfile =  (
+        <div>
+            EditProfile: Please wait for USER
+        </div>
+    )
+    //As soon as User(almost instantly) render correct EditPlayerProfile View
+    if(user){
+         editProfile  = (
+             <div className="userPage container">
+                 <ul className="userPage player-info-container">Current Username: {user.username}</ul>
+                 <FormField
+                     label="New Username"
+                     value={username}
+                     onChange={un => setUsername(un)}
+                 />
+                 <ul className="userPage player-info-container">Current Gender:{user.gender} </ul>
+                 <ul className="userPage player-info-container">
+                     <Select
+                         options={genderOptions}
+                         onChange={(genders)=>setGender(genders.value)}
+                     />
+                 </ul>
+                 <div className="UserPage button-container">
+                     <Button
+                         width="100%"
+                         onClick={() => doEdit()}
+                     >
+                         Accept Changes
+                     </Button>
+                 </div>
+             </div>
+        )
+    }
+
+    const doEdit = async () => {
+        try {
+            localStorage.removeItem('errorMessage');
+            setUsername(username.trim())
```

Comment:
null does not have a trim() method (error when press Save without new username)

Reply:
Username is never null, the username is the old username, so if you dont select a new one, the old one just gets passed